api/src/main/java/me/lokka30/treasury/api/economy/EconomyProvider.java

```diff
@@ -140,6 +142,36 @@
      */
     void retrieveBankAccountIds(@NotNull EconomySubscriber<Collection<UUID>> subscription);
 
+    /**
+     * Request all {@link BankAccount bank accounts} the given player is a member of.
+     *
+     * @param playerId     the player
+     * @param subscription the {@link EconomySubscriber} accepting the resulting value
+     * @author MrNemo64
+     * @see #retrieveAllBankAccountsPlayerHasPermission(UUID, BankAccountPermission[], EconomySubscriber)
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    void retrieveAllBankAccountsPlayerIsMemberOf(
+            @NotNull UUID playerId,
+            @NotNull EconomySubscriber<Collection<UUID>> subscription
+    );
+
+    /**
+     * Request all the {@link BankAccount bank accounts} where the given player has the given permissions.
+     *
+     * @param playerId     the player
+     * @param permissions  the permissions that the given player has to have on the {@link BankAccount account}
+     * @param subscription the {@link EconomySubscriber} accepting the resulting value
+     * @author MrNemo64
+     * @see #retrieveAllBankAccountsPlayerIsMemberOf(UUID, EconomySubscriber)
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    void retrieveAllBankAccountsPlayerHasPermission(
```

Comment:
In my opinion this should be UUID, EconomySubscriber, BankAccountPermission... because it allows you to do stuff as

```java
economyProvider.retrieveAllBankAccountsPlayerHasPermission(playerId, subscription, BankAccountPermision.FOO, BankAccountPermission.BAR);
```

Reply:
I thought about that but then the arguments get splitted into multiple lines. TBH I also prefer this way of doing it, if @Jikoo and @lokka30 can give their opinions

Reply:
I do also prefer varargs for variable length array arguments.